### PR TITLE
Add Agency Profile Form Page

### PR DESCRIPTION
**Pull request by emmanuelclflores**
### Administrative Info

Monday Board ID: 851212620
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

Added Agency Profile Form page and basic Jest unit testing.

### Testing

How did you confirm your changes worked?

Ran npm test: all tests passed.

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Agency Profile Form Page  |
| ----------------------- |
image

### CODE REVIEW

**Review by sblumin — CHANGES_REQUESTED** [review]
Looks good! Please add file headers to your different jsx files to describe what the file is/what it does/why. Also add comments to your test file to describe the different tests being performed.

**Comment by wmwuu** [comment]
Documentation added :o

**Review by sblumin — APPROVED** [review]
LGTM! Great job adding comments they look beautiful

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,9 @@
-import logo from './logo.svg';
-import './App.css';
+import logo from "./logo.svg";
```
Comment (nicolaswl) [inline src/App.js:1]: This is never used

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,9 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+  return <AgencyProfileForm />;
```
Comment (nicolaswl) [inline src/App.js:6]: We need to make the profile page a link through the router.

File: src/pages/agency-profile-form/CheckboxColumn.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * Expected props:
+ * - {Array<Object>} options: list of objects containing data about each
+ * checkbox option (title and whether it is currently selected)
+ * - {Number} indexBuffer: value to add to indices (accounts for multiple
+ * columns)
+ * - {Function} onChange: callback to handle input changes, should take a
+ * Number
+ */
+export default function CheckboxColumn(props) {
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/CheckboxColumn.jsx:18]: We should probably be consistent on two points here:

1. exporting functions at the bottom
2. declaring components as classes vs functions (typically function components have no state)

File: src/pages/agency-profile-form/AddressList.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * AddressList encapsulates the expandable list of additional addresses in the
+ * agency form. It generates a series of InputText components corresponding to
+ * each element in the addresses list.
+ *
+ * Expected props:
+ * - {Array<String>} addresses: list of addresses
+ * - {String} stateKey: key to pass into the onChange callback
+ * - {Function} onChange: callback from the form page to handle input changes,
+ * should take a String and an Any
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/AddressList.jsx:14]: should be String and an Array I think

File: src/pages/agency-profile-form/AgencyProfileForm.js [hunk 5]
```diff
@@ -0,0 +1,860 @@
...
+        primaryContact: "",
+        mainSiteAddress1: "",
+        agencyStatus: "",
+        region: "",
+        sanDiegoDistrict: "",
+        countyDistrict: "",
+        stateAssemblyDistrict: "",
+        stateSenateDistrict: "",
+        federalCongressionalDistrict: "",
+        mainSiteAddress2: "",
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/AgencyProfileForm.js:44]: Why do we ask this twice on the form?

File: src/pages/agency-profile-form/AgencyProfileForm.js [hunk 6]
```diff
@@ -0,0 +1,860 @@
...
+
+          <div className="form-section">
+            <Row noGutters={true}>
+              <Col>
+                <button
+                  type="button"
+                  className="form-button-submit"
+                  onClick={this.submitForm}
+                >
+                  Save Profile
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/AgencyProfileForm.js:840]: This shows as Create Profile on the Figma, will need to change to Save on editing

File: src/pages/agency-profile-form/AgencyProfileForm.js [hunk 7]
```diff
@@ -0,0 +1,860 @@
...
+                  subkey: "immigrant",
+                },
+              ]}
+              stateKey="demographicType"
+              onChange={this.handleInputChange}
+              twoColumns
+            />
+          </div>
+
+          <div className="form-section">
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/AgencyProfileForm.js:810]: We should clarify what type of input "Assigned Staff" is meant to be

File: src/pages/agency-profile-form/Checkbox.jsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+ * - {Boolean} isChecked: whether it should be currently checked
+ * - {String} label: text to display next to the checkbox
+ * - {Number} index: index to pass into the onChange callback
+ * - {Function} onChange: callback to handle input changes, should take a
+ * Number
+ */
+class Checkbox extends Component {
+  constructor(props) {
+    super(props);
+    //this.state = { isChecked: props.isChecked };
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/Checkbox.jsx:18]: delete

File: src/pages/agency-profile-form/Checkbox.jsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  /**
+   * Callback function to handle toggling the checkbox. Passes the index prop
+   * up to the onChange callback.
+   */
+  onChange = () => {
+    const { index, onChange } = this.props;
+    /*this.setState((prevState) => ({
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/Checkbox.jsx:27]: delete

File: src/pages/agency-profile-form/FormSectionHeader.jsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import "./formstyle.css";
+
+/**
+ * Header for each section in the agency form.
+ * Expected props:
+ * - {String} title: title of the section
+ */
+export default function FormSectionHeader(props) {
```
Comment (nicolaswl) [inline src/pages/agency-profile-form/FormSectionHeader.jsx:9]: export and class component consistency

**Review by nicolaswl — COMMENTED** [review]
Great Job! I added some simple feedback and recommendations for changes that should be looked at before merging.

File: src/pages/agency-profile-form/FormSectionHeader.jsx (hunk 10 above)
Comment (emmanuelclflores) [reply src/pages/agency-profile-form/FormSectionHeader.jsx]: Resolved

File: src/pages/agency-profile-form/CheckboxColumn.jsx (hunk 3 above)
Comment (emmanuelclflores) [reply src/pages/agency-profile-form/CheckboxColumn.jsx]: Shifted everything to class-based components

**Comment by wmwuu** [comment]
Resolved CircleCI issues and merge conflicts :-)

**Review by nicolaswl — APPROVED** [review]

**Pull request merged**